opensora/datasets/read_video.py

```diff
@@ -169,47 +169,14 @@ def _read_from_stream(
         return []
 
     # == main ==
-    buffer_count = 0
-    frames_pts = []
-    cnt = 0
     try:
-        for _idx, frame in enumerate(container.decode(**stream_name)):
-            frames_pts.append(frame.pts)
-            video_frames[cnt] = frame.to_rgb().to_ndarray()
-            cnt += 1
-            if cnt >= len(video_frames):
-                break
-            if frame.pts >= end_offset:
-                if should_buffer and buffer_count < max_buffer_size:
-                    buffer_count += 1
-                    continue
+        for idx, frame in enumerate(container.decode(**stream_name)):
+            if idx >= len(video_frames):
                 break
+            video_frames[idx] = torch.from_numpy(frame.to_rgb().to_ndarray())
```

Comment:
note for future, there are a ton of temporary values created here. potentially explicit delete of them